packages/storage/amplify_storage_s3/lib/src/S3PrefixResolver/prefix_handler.dart

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/amplify_core.dart';
+
+import '../types.dart';
+
+class PrefixHandler {
+  static Future<dynamic> handlePrefix(Map<String, dynamic> arguments,
+      AmplifyStorageS3PrefixResolver prefixResolver) async {
+    final accessLevelString = arguments['accessLevel'];
+    StorageAccessLevel accessLevel;
+    switch (accessLevelString) {
+      case 'public':
+        // Why are our storageAccessLevel enums different?
```

Comment:
The default prefix for `StorageAccessLevel.guest` is always `"public"` I think, do you see any inconsistency in the native libraries?

Reply:
The StorageAccessLevel enum is not the same between platforms which requires the extra case in the switch statement as a result.  But yes they map to the same S3 concepts. 

iOS: 
```
public enum StorageAccessLevel: String {

    /// Objects can be read or written by any user without authentication
    case guest

    /// Objects can be viewed by any user without authentication, but only written by the owner
    case protected

    /// Objects can only be read and written by the owner
    case `private`
}
```

Android: 
```
public enum StorageAccessLevel {

    /**
     * Storage items are accessible by all users of your app.
     */
    PUBLIC,

    /**
     * Storage items are readable by all users, but writable only by the creating user.
     */
    PROTECTED,

    /**
     * Storage items are accessible for the individual user who performs the write.
     */
    PRIVATE
}
```

Reply:
iOS enum is `guest`, but it looks like that [the prefix value it returns should always be `"public"`](https://github.com/aws-amplify/amplify-ios/blob/c6d870a837c7765439fc06a31ba2aa1472913f05/AmplifyPlugins/Storage/AWSS3StoragePlugin/Support/Utils/StorageRequestUtils%2BGetter.swift#L64).